pkg/policy/k8s/cilium_network_policy.go

```diff
@@ -84,7 +55,7 @@ func (k *K8sWatcher) ciliumNetworkPoliciesInit(ctx context.Context, cs client.Cl
 		cnpCache := make(map[resource.Key]*types.SlimCNP)
```

Comment:
Let's lift these caches to a field on the PolicyWatcher. That should simplify method signatures nicely.